packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -149,6 +149,20 @@ export interface MathExpressionOptions {
    * @default Duration.minutes(5)
    */
   readonly period?: cdk.Duration;
+
+  /**
+   * Account to evaluate the expression within (for example when an expression includes SEARCH or SERVICE_QUOTA)
+   *
+   * @default - Deployment account.
```

Comment:
For my understanding:

This will ONLY affect math expressions that are NOT based on underlying real metrics, right?

As in, if I have a math expression that does something like `M1 / M2` (given real metric streams), and for example `M1` and `M2` would both come from account `1111`, but I specify `2222` here on the math expression, that would have no impact?

If so, I would like to see that stated a little more clearly in the docstrings here.

Reply:
Can you please also add a section to the README that describes the practices around cross-account/cross-region metrics?

Reply:
> if I have a math expression that does something like M1 / M2 (given real metric streams), and for example M1 and M2 would both come from account 1111, but I specify 2222 here on the math expression, that would have no impact?

Yes, I believe that is correct, although if an expression contained both real metric streams, and a search expression (for example `M1 / SEARCH('{"Some Namespace"}', 'Average', 300)`, then `M1` would come from whichever account is specified on the metric (or the deployment account), and the `SEARCH` results would be from the account specified on the expression. In practice, I'm not sure how common this use case is.

Additionally, it turns out cross-account SERVICE_QUOTA retrieval is not supported, but cross-region is (on the same account). I'd be interested to get your thoughts on how we should handle this. Few thoughts as to options on my side:

1. We could separate SEARCH and SERVICE_QUOTAs entirely from `MathExpression`s and apply the additional cross-env constraints to them only. Using `MathExpression` for the SEARCH use case does seem to have caused some confusion already ([#7237](https://github.com/aws/aws-cdk/issues/7237)), and it may simplify the user-facing docs and props (not requiring `usingMetrics` for SearchExpressions for example, although it may also make the use case mentioned above more complex),
2. We could keep this change largely as is, but parse the expression to check if it contains a cross-account SERVICE_QUOTA expression function,
3. Not handle the SERVICE_QUOTA constraints on the assumption that the use case is very uncommon and stack deployment will fail rather than a CDK error. Not an ideal CX, although it may be something that support is added for a later date anyway.

Reply:
3 seems simplest to me, and I'm partial to removing `usingMetrics` as well.

I guess the alternative might be to have `IMetric`-like objects that represent the `SEARCH()` expression itself:

```ts
new MathExpression('m1', {
  usingMetrics: {
    m1: new SearchExpression('...', { account: '1234' }),
  },
})
```

OO-wise this is probably a good idea, but will somewhat complicate implementations: you'll need to refer to it as `m1` in your expression, and then the implementation would need to be aware that it needs to substitute `m1` with the actual `SEARCH()` expression when rendering. Also, it will look VERY different from what users who are used to the CloudWatch expression language will be used to, and therefore foreign and will need a lot of explanation.

It's also not a blocker to now just do the simple thing, and do the more "correct", complicated thing later.

So I'm partial to adding `account`, `region`, `period` on MathExpression itself and making `usingMetrics` optional, as long as it's very clear that the new fields only apply to non-metric expressions. Maybe we can even call them `searchAccount`, `searchRegion`, `searchPeriod`...

Reply:
I had been considering adding a `SearchExpression` class that extends (or is otherwise similar to) `MathExpression` so that the differences could be documented more clearly, and it would allow for more granular error messages (such as "Search expressions cannot be used to create Alarms"), something like:

```js
new GraphWidget({
   left: [
      new SearchExpression({
         expression: "SEARCH(...) + m1",
         account: "1111",
         region: "us-east-1",
         usingMetrics: { // defaults to {}
            m1: ...
         },
      }),
   ],
});
```

however maybe that detaches things too far (since it would still be possible to use the `SEARCH` function within `MathExpression`). I think the current confusion around implementing search expressions in CDK is that the [CloudWatch docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/search-expression-syntax.html) don't really say that search expressions are just math expressions - instead they split out math expressions and search expressions into their own page (and there's no mention of math expressions on the search expression page).

I'm happy to go with option 3 and expand on the existing MathExpression docs to make clear that they can be used for search expressions also, and to go ahead with the proposed interface changes you've suggested.